core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -245,9 +245,7 @@ struct HIPParallelLaunchKernelFunc<DriverType, Kokkos::LaunchBounds<0, 0>,
       HIPParallelLaunchKernelFuncData<DriverType, Kokkos::LaunchBounds<0, 0>,
                                       HIPLaunchMechanism::LocalMemory>;
   static auto get_kernel_func() {
-    return HIPParallelLaunchKernelFunc<
-        DriverType, Kokkos::LaunchBounds<HIPTraits::MaxThreadsPerBlock, 1>,
-        HIPLaunchMechanism::LocalMemory>::get_kernel_func();
+    return hip_parallel_launch_local_memory<DriverType>;
```

Comment:
In that case, shouldn't we get rid of that specialization altogether and just change `default_launchbounds()` in the primary template to return something like `MaxThreadsPerBlock == 0 && MinBlocksPerSM == 0`?

(Haven't looked whether they are further simplification opportunities)

Reply:
You mean the primary template returns `hip_parallel_launch_local_memory<DriverType, MaxThreadsPerBlock, MinBlocksPerSM>` instead?
OK I think I see your point.

Reply:
Right and in `hip_parallel_launch_local_memory<DriverType, MaxThreadsPerBlock, MinBlocksPerSM>`, we have `__launch_bounds__(maxTperB, minBperSM)` as part of the signature which we don't have for  `hip_parallel_launch_local_memory<DriverType>` .